.github/README.md

```diff
@@ -0,0 +1,24 @@
+# STLab CI Configuration
+
+The CI configuration in `workflows/build_and_test.yml` strives to be a minimal wrapper around the root-level `stlab.py` script, which will be invoked thrice (once each for `install`, `build`, and `test` subcommands) for each configuration listed in `.matrix.json`.
+
+Please avoid adding platform-specific build configuration directly in `build_and_test.yml` - this makes it difficult for developers to reliably reproduce our build configuration on a new machine. Instead, add the necessary platform-specifics to the root-level `stlab.py`, so the CI system, and our developers, can benefit from the change. 
```

Comment:
I'd suggest removing this paragraph as it is redundant with above and comes across as a bit patronizing.